microsoft/PowerToys
# [Installer] Restart explorer.exe programatically after successful update

## Summary of the Pull Request
Closing/restarting explorer.exe is required during uninstall / upgrade procedure in order for new dll's for `PowerRename` and `ImageResizer` to be reloaded (and old ones unloaded). Closing explorer.exe by force during uninstallation leaves bad user experience.

Show message box on `PowerToys` start (after successful update) and ask users if they want to (gracefully) restart explorer now (performed by `PowerToys` using `RestartManager` API that will preserve open Explorer windows) or they can do it later.

![image](https://user-images.githubusercontent.com/57061786/84167085-11d11780-aa76-11ea-82ae-05d816fbead2.png)



## PR Checklist
* [x] Applies to #2730 https://github.com/microsoft/PowerToys/issues/2012


## Validation Steps Performed
Start runner with `-report_update_success` command line argument (equivalent of starting runner after successful update).
Expected result: message box appears, and if `yes` is selected, explorer is restarted successfully and `PowerToys` start.
